### ISSUE TITLE
KUTE.js anyone?

### ISSUE THREAD

**Issue by thednp** [haltu/muuri#99]
Interested in a KUTE.js integration? If so, what would be the requirements?

**Comment by niklasramo** [comment]
I don't know if you have checked out what's happening in the [dev branch](https://github.com/haltu/muuri/tree/dev), but I've already refactored Muuri to use Web Animations API after much experimentation with various animation libraries, CSS transitions and of course Web Animations API. I'm really happy with the performance of Web Animations (even with the [polyfill](https://github.com/web-animations/web-animations-js)).

However, if you're into building a "plugin" that would replace the Web Animations API with KUTE.js I'm more than happy to help out =)

**Comment by niklasramo** [comment]
Muuri's API is designed in a way that makes overwriting the internal animation engine possible, but it's not trivial.

**Comment by thednp** [comment]
Thanks for the feedback. I am now thinking why isn't this working with basic CSS3 transitions after all, while leaving hooks open to animation engines for more animation control.

Older muuri.js is using Velocity which was pretty good but a bit too much for an animated grid, that's why I suggested my library KUTE.js, it's a bit more modular and over performs a lot of other animation engines, including Velocity.

**Comment by niklasramo** [comment]
Yep, CSS transitions were my initial choice back in the day when I started working on this lib, but I never got them working as I wanted (it's a long story). So I chose Velocity instead and everything worked perfectly.

However, I kept on benchmarking different open source animation libraries, but never saw any perf gains compared to Velocity. Then one day I tested web animations (although they have lousy support) and they seemed to be super slick in my tests. Additionally, web animations are much more easier to control via JavaScript than CSS transitions and they seem to be the future standard for animating things on web. The only problem is that web animations are not so well supported, but thankfully there exists an awesome [polyfill](https://github.com/web-animations/web-animations-js/tree/master) that worked beautifully in my tests. So, one thing led to another and I decided to replace Velocity with web animations. I actually released yesterday the 0.5.0 version which has web animations.

The hooks are still open for other animation engines if you want to experiment with KUTE.js. And actually Muuri's current API allows only pretty simple animation customizations so would be cool if there was a way to do some _crazier_ animations 😜
